### PR TITLE
Proxy pipeline refactoring part 1 

### PR DESCRIPTION
**Pull request by basilisk487**

### CODE REVIEW

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java [hunk 1]
```diff
@@ -233,47 +229,78 @@ void setMetric(String value) {
...
+    }
+
+    @NotNull
+    Map<String, String> getAnnotationMap() {
+      return annotations;
+    }
+  }
+
+  /**
+   * This class provides a wrapper around ReportPoint.
```
Comment (vikramraman) [inline java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java]: wrapper around Span..

File: java-lib/src/main/java/com/wavefront/ingester/ReportPointDecoderWrapper.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * Wraps {@link Decoder} as {@link ReportableEntityDecoder}.
+ *
+ * @author vasily@wavefront.com
+ */
+public class ReportPointDecoderWrapper implements ReportableEntityDecoder<String, ReportPoint> {
+
+  private final Decoder<String> delegate;
+
+    public ReportPointDecoderWrapper(@NotNull Decoder<String> delegate) {
```
Comment (vikramraman) [inline java-lib/src/main/java/com/wavefront/ingester/ReportPointDecoderWrapper.java]: Formatting appears off. 2 vs 4 spaces.

File: java-lib/src/main/java/com/wavefront/ingester/SpanIngesterFormatter.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          case "traceId":
+            span.setTraceId(annotation.getValue());
+            iterator.remove();
+            break;
+          default:
+            break;
+        }
+      }
+    }
+
```
Comment (vikramraman) [inline java-lib/src/main/java/com/wavefront/ingester/SpanIngesterFormatter.java]: The span name cannot be empty either.

Comment (vikramraman) [reply java-lib/src/main/java/com/wavefront/ingester/SpanIngesterFormatter.java]: Nvm. Guess the Formatter handles that scenario.

File: proxy/src/main/java/com/wavefront/agent/handlers/AbstractReportableEntityHandler.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+  protected long getReceivedFiveMinuteRate() {
+    return receivedStats.stream().mapToLong(i -> i).sum() / 300;
+  }
+
+  protected SenderTask getTask() {
+    return senderTasks.get((int)(roundRobinCounter.getAndIncrement() % senderTasks.size()));
+  }
+  /*
+  protected SenderTask getTask() {
```
Comment (vikramraman) [inline proxy/src/main/java/com/wavefront/agent/handlers/AbstractReportableEntityHandler.java]: Remove?

File: proxy/src/main/java/com/wavefront/agent/handlers/SpanHandlerImpl.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    this.deliveredCounter = Metrics.newCounter(new MetricName("traces." + handle, "", "delivered"));
+
+    this.statisticOutputExecutor.scheduleAtFixedRate(this::printStats, 10, 10, TimeUnit.SECONDS);
+    this.statisticOutputExecutor.scheduleAtFixedRate(this::printTotal, 1, 1, TimeUnit.MINUTES);
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  protected void reportInternal(Span span) {
+    //validatePoint(span, handle, Validation.Level.NUMERIC_ONLY);
```
Comment (vikramraman) [inline proxy/src/main/java/com/wavefront/agent/handlers/SpanHandlerImpl.java]: remove?

File: java-lib/src/main/java/com/wavefront/ingester/SpanIngesterFormatter.java (hunk 3 above)
Comment (basilisk487) [reply java-lib/src/main/java/com/wavefront/ingester/SpanIngesterFormatter.java]: @vikramraman everything else has been addressed

**Pull request merged**